spec/models/ext_management_system_spec.rb

```diff
@@ -254,10 +254,23 @@
     end
   end
 
-  context "with virtual totals" do
+  context "with virtual totals and virtual aggregate" do
     before(:each) do
       @ems = FactoryGirl.create(:ems_vmware)
-      (1..2).each { |i| FactoryGirl.create(:vm_vmware, :ext_management_system => @ems, :name => "vm_#{i}") }
+      (1..2).each { FactoryGirl.create(:vm_vmware, :ext_management_system => @ems) }
+      (1..2).each do
+        FactoryGirl.create(:host,
```

Comment:
this works.

If it works better for you, there is also `FactoryGirl.create_list(:host, 2, :ext_management_system => @ems, ...`

The old version required upon `i` for `:name`, but since it is not needed, `create_list` probably simplifies things for you.